inductiva/resources/machines.py

```diff
@@ -54,66 +58,76 @@ def start(self):
                 disk_size_gb=self.disk_size_gb,
                 zone=self.zone,
             )
-            # body.resource_pool_id = self.machine_group_id
             try:
                 logging.info("Creating a machine group. \
                              This may take a few minutes.")
-                api_instance.create_instance_group(body=body)
+                start_time = time()
+                api_instance.create_instance_group(body=instance_group_config)
+                self.estimated_cost = self._compute_estimated_cost(api_instance)
+                creation_time = (time() - start_time) / 60
                 logging.info(
                     "Machine group with the specified settings successfully \
                         created.")
-                logging.info("Machine type: %s", self.machine_type)
-                logging.info("Number of machines: %s", self.num_machines)
-                logging.info("Spot: %s", self.spot)
-                logging.info("Disk size: %s GB", self.disk_size_gb)
-                logging.info("Machine group ID: %s", self.machine_group_id)
-                if self.name:
-                    logging.info("Name: %s", self.name)
-                logging.info("Machine group successfully created.")
-
-            except ApiException as e:
-                raise e
-        return self.machine_group_id
+                self._log_machine_group_info()
+                logging.info("Machine group successfully created in %s mins.",
+                             creation_time)
+
+            except inductiva.client.ApiException as api_exception:
+                raise api_exception
 
     def terminate(self):
+        """Terminates a machine group."""
 
-        with ApiClient(self.api_config) as client:
-            api_instance = InstanceApi(client)
+        with inductiva.client.ApiClient(self.api_config) as client:
```

Comment:
why are we constantly creating a client. Couldn't this be created at the time of the construction? 
Do clients have timeout? I really don't know.
But if a client is not something that has a time to live, why are we constantly creating a client?